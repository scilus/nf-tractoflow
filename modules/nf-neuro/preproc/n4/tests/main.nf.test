nextflow_process {

    name "Test Process PREPROC_N4"
    script "../main.nf"
    process "PREPROC_N4"

    tag "modules"
    tag "modules_nfcore"
    tag "preproc"
    tag "preproc/n4"

    tag "subworkflows"
    tag "subworkflows/load_test_data"

    setup {
            run("LOAD_TEST_DATA", alias: "LOAD_DATA") {
                script "../../../../../subworkflows/nf-neuro/load_test_data/main.nf"
                process {
                    """
                    input[0] = Channel.from( [ "heavy.zip" ] )
                    input[1] = "test.load-test-data"
                    """
                }
            }
        }

    test("preproc - n4 - dwi") {
        config "./nextflow_dwi.config"
        when {
            process {
                """
                input[0] = LOAD_DATA.out.test_data_directory
                    .map{ test_data_directory -> [
                    [ id:'test', single_end:false ], // meta map
                    file("\${test_data_directory}/dwi_with_b0/dwi.nii.gz", checkIfExists: true),
                    file("\${test_data_directory}/dwi_with_b0/b0.nii.gz", checkIfExists: true),
                    file("\${test_data_directory}/dwi_with_b0/mask.nii.gz", checkIfExists: true)]}
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("preproc - n4 - anat") {
        when {
            process {
                """
                input[0] = LOAD_DATA.out.test_data_directory
                    .map{ test_data_directory -> [
                    [ id:'test', single_end:false ], // meta map
                    file("\${test_data_directory}/anat/anat_image.nii.gz", checkIfExists: true),
                    [],
                    []]}
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
