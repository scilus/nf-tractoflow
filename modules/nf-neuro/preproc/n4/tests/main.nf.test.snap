{
    "preproc - n4 - dwi": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__image_n4.nii.gz:md5,6383b9ffb68d04bf089ba7a040c4a77f"
                    ]
                ],
                "1": [
                    "versions.yml:md5,08cced153fb00b483f3a59216e05aaff"
                ],
                "image": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__image_n4.nii.gz:md5,6383b9ffb68d04bf089ba7a040c4a77f"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,08cced153fb00b483f3a59216e05aaff"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-10T16:06:45.886061"
    },
    "preproc - n4 - anat": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__image_n4.nii.gz:md5,a0329a81f3cc26dfa04e3fb8c6e6dc66"
                    ]
                ],
                "1": [
                    "versions.yml:md5,08cced153fb00b483f3a59216e05aaff"
                ],
                "image": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__image_n4.nii.gz:md5,a0329a81f3cc26dfa04e3fb8c6e6dc66"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,08cced153fb00b483f3a59216e05aaff"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-10T16:09:11.896134"
    }
}